### ISSUE TITLE
Extend the logger and document it

### ISSUE THREAD

**Issue by jacobcook1995** [ImperialCollegeLondon/virtual_rainforest#105]
**Is your feature request related to a problem? Please describe.**
At the moment the logger is not documented, mainly because what it is currently doing is very simple. As we develop v0.1 we will need to use more complex logging behaviours (e.g. nesting logging entries). This extension will need to be documented so that all developers know how to use it correctly/consistently.

**Describe the solution you'd like**
When the logger is extended documentation really must be added.

**Comment by jacobcook1995** [comment]
We've basically decided that more complex logging behaviour isn't required for v0.1, so this issue can be closed once the documentation improvement from #115 is merged into develop
